mtags/src/main/scala-3/scala/meta/internal/pc/completions/MatchCaseCompletions.scala

```diff
@@ -86,34 +86,42 @@ object CaseKeywordCompletion:
             new Parents(argTypes, definitions)
           case _ =>
             new Parents(NoType, definitions)
-      case sel => new Parents(sel.tpe, definitions)
+      case sel =>
+        val selectorTpe = sel.tpe
+        new Parents(selectorTpe, definitions)
 
     val selectorSym = parents.selector.typeSymbol
-
     // Special handle case when selector is a tuple or `FunctionN`.
     if definitions.isTupleClass(selectorSym) || definitions.isFunctionClass(
         selectorSym
       )
     then
+      val selectorTpe = parents.selector.show
+      val tpeLabel =
+        if !selectorTpe.contains("x$1") then selectorTpe
```

Comment:
Maybe it'd be better to match on the `parents.selector` to check when there will be an argument? In case somehow it can have a different name than `x$1`.